docs/how-tos/protocol.rst

```diff
@@ -2,3 +2,541 @@
 
 How to define a new ``Protocol``
 ================================
+
+The **gufe** :ref:`Protocol <protocol>` system is designed as an *extensible point* of the library,
+allowing developers to write their own methods for performing free energy calculations in a form that can take full advantage of the OpenFE ecosystem.
+
+Our recommendation in this how-to is to start simple;
+we will call out areas where you can iterate further on making your ``Protocol`` more sophisticated.
+
+.. image:: ../_static/gufe_protocol_diagram.svg
+    :alt: The ``gufe`` protocol system.
+
+
+Overview: What makes a ``Protocol``
+-----------------------------------
+
+A complete ``Protocol`` implementation requires the following interconnected components:
+
+1. **Protocol class**: The main class that creates computational workflows
+2. **Settings class**: A `Pydantic <https://docs.pydantic.dev/latest/>`_ model defining configuration parameters
+3. **ProtocolUnit classes**: Individual computational steps in the workflow
+4. **ProtocolResult class**: Container for aggregated results from multiple runs
+
+The ``Protocol`` creates a :ref:`ProtocolDAG <protocoldag>` (directed acyclic graph) of :ref:`ProtocolUnit <protocolunit>` objects that define the computational workflow.
+Multiple ``ProtocolDAG`` executions can be aggregated into a single :ref:`ProtocolResult <protocolresult>` to provide statistical estimates.
+
+If this is your first time writing a ``Protocol``, we recommend defining these components in the following step order.
+You can of course iterate on these components as you write each one, improving the implementation as you go.
+
+
+Step 1: Define your Settings
+-----------------------------
+
+First, create a settings class that inherits from :class:`.Settings`.
+This defines all the configuration parameters your protocol needs:
+
+.. code-block:: python
+
+    from gufe.settings import Settings
+    import pydantic
+    from openff.units import unit
+
+    class MyProtocolSettings(Settings):
+        """Settings for an example Protocol.
+
+        """
+        # the number of independent simulations to perform
+        n_repeats: int = 5
+        
+        simulation_length: pydantic.Field(
```

Comment:
I think we should add more details on the unit typing system we have here and introduce things like `FloatQuantity["nanosecond"]`, something like for settings which require units you can use FloatQuantity and provide a default unit which the value will be converted to this also ensures that the provided unit is compatible with what you intend.

Reply:
as discussed in gufe dev call, `FloatQuantity` will no longer be used as we move to vendoring interchange and using pydantic v2.

Reply:
I've updated the `Settings` example to use the new typing system!